### PR TITLE
Move the examples into separate folders

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Main advantage:
- Obvious for users -> one folder = one example

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
This approach increased CI times by 50%.
Will try another approach

**Comment by Hofer-Julian** [comment]
The examples are now in separate folders, with the locations specified in Cargo.toml

I've also added a Readme with screenshot per example and fixed the links in the top Readme.

File: examples/README.md [hunk 1]
```diff
@@ -2,19 +2,18 @@
...
+- [Custom Editable](./custom_editable/main.rs)
+- [Custom Orientable](./custom_orientable/main.rs)
+- [Custom Paintable](./custom_paintable/main.rs)
+- [Custom Widget](./custom_widget/main.rs)
+- [Dialog](./dialog/main.rs)
+- [Entry Completion](./entry_completion/main.rs)
+- [Entry Undo](./entry_undo/main.rs)
+- [GtkBuilder example](./gtk_builder/main.rs)
+- [TextView](./text_viewer/main.rs)
+- [Video Player](./video_player/main.rs)
```
Comment (bilelmoussaoui) [inline examples/README.md:19]:
```suggestion
- [Basic example](./basics/)
- [Using the Builder pattern](./builder_pattern/)
- [Clock example](./clock/)
- [Composite Template](./composite_template/)
- [CSS](./css/)
- [Custom Editable](./custom_editable/)
- [Custom Orientable](./custom_orientable/)
- [Custom Paintable](./custom_paintable/)
- [Custom Widget](./custom_widget/)
- [Dialog](./dialog/)
- [Entry Completion](./entry_completion/)
- [Entry Undo](./entry_undo/)
- [GtkBuilder example](./gtk_builder/)
- [TextView](./text_viewer/)
- [Video Player](./video_player/)
```
No need to point to the source code, there are README's for each example now

Comment (Hofer-Julian) [reply examples/README.md:19]: I was wondering which would better, if you are more sure let's point to the READMEs :)

**Comment by bilelmoussaoui** [comment]
Looks good to me, one thing I have been wondering about, can we use multiple files per example ? if that's the case, would you mind splitting the custom editable's `CustomTag` to a separate file ?

**Comment by Hofer-Julian** [comment]
> Looks good to me, one thing I have been wondering about, can we use multiple files per example ? if that's the case, would you mind splitting the custom editable's `CustomTag` to a separate file ?

Sure, will do that!

**Comment by sdroege** [comment]
Should do the same for `gtk-rs`. Can you open an issue for that, or even better a PR?

**Comment by Hofer-Julian** [comment]
> Should do the same for `gtk-rs`. Can you open an issue for that, or even better a PR?

Will open a PR. Not that I have anything better to do on a 4.5h train ride :P

**Comment by Hofer-Julian** [comment]
@bilelmoussaoui separated custom_editable and squashed the commits.

**Comment by bilelmoussaoui** [comment]
Oh no, random clippy changes :P Will fix it in a separate PR & then we can rebase & merge this one.

**Comment by Hofer-Julian** [comment]
Thanks!
Can you explain me why clippy is failing now, because I don't understand it^^

**Comment by bilelmoussaoui** [comment]
> Thanks!
> Can you explain me why clippy is failing now, because I don't understand it^^

A change on clippy made a lint enabled by default I guess? and it's affecting the automatically generated sys (the 1:1 C API in Rust)'s lib.rs. This patch https://github.com/gtk-rs/gir/pull/1082 is needed to fix it.

**Comment by bilelmoussaoui** [comment]
This can be rebased now

**Comment by Hofer-Julian** [comment]
> This can be rebased now

Done.

**Comment by bilelmoussaoui** [comment]
Thanks Julian!

**Comment by YaLTeR** [comment]
Seems like this PR dropped the application subclass example/test from https://github.com/gtk-rs/gtk4-rs/pull/228.

**Comment by Hofer-Julian** [comment]
Good catch.
Seems to be one of the examples where git rebase doesn't do what you want.

**Comment by bilelmoussaoui** [comment]
Do you mind sending a PR to re-add it please? :-)

**Comment by Hofer-Julian** [comment]
Just submitted ;-)

**Pull request merged**